pacai/core/distanceCalculator.py

```diff
@@ -54,103 +42,58 @@ def getDistanceOnGrid(self, pos1, pos2):
 
         raise Exception("Position not in grid: " + str(key))
 
-    def isReadyForMazeDistance(self):
-        return (self._distances is not None)
-
-def isInt(pos):
-    x, y = pos
-    return x == int(x) and y == int(y)
-
-def getGrids2D(pos):
-    grids = []
-    for x, xDistance in getGrids1D(pos[0]):
-        for y, yDistance in getGrids1D(pos[1]):
-            grids.append(((x, y), xDistance + yDistance))
-    return grids
-
-def getGrids1D(x):
-    intX = int(x)
-    if x == int(x):
-        return [(x, 0)]
-    return [(intX, x - intX), (intX + 1, intX + 1 - x)]
-
-##########################################
-# MACHINERY FOR COMPUTING MAZE DISTANCES #
-##########################################
-
-distanceMap = {}
-
-class DistanceCalculator:
-    def __init__(self, layout, distancer):
-        self.layout = layout
-        self.distancer = distancer
-        self.cache = {}
-
-    def run(self):
-        if self.layout.walls not in self.cache:
-            self.cache[self.layout.walls] = computeDistances(self.layout)
-
-        self.distancer._distances = self.cache[self.layout.walls]
-
-def computeDistances(layout):
-    """
-    Runs UCS to all other positions from each position.
-    """
-
-    distances = {}
-    allNodes = layout.walls.asList(False)
-
-    for source in allNodes:
-        dist = {}
-        closed = {}
-
-        for node in allNodes:
-            dist[node] = sys.maxsize
-
-        queue = priorityQueue.PriorityQueue()
-        queue.push(source, 0)
-        dist[source] = 0
```

Comment:
You don't need to implement a round2D function. Python has a built in round function. e.g. round(5.51) will result in 6. No need to do this whole function.